src/main/resources/hazelcast-config-schema.json

```diff
@@ -1373,22 +1373,129 @@
+                    "MEGABYTES",
+                    "GIGABYTES"
+                  ],
+                  "default": "MEGABYTES",
+                  "description": "Unit can be bytes, kilobytes, megabytes and gigabytes. Default unit is MEGABYTES."
+                },
+                "value": {
+                  "type": "number",
+                  "default": 128,
+                  "minimum": 1,
+                  "description": "Default value is 128."
+                }
+              },
+              "description": "Size of the total native memory to allocate. You can give the size using its \"value\" attribute and the unit using its \"unit\" attribute."
+            },
+            "min-block-size": {
+              "type": "integer",
+              "default": 16,
+              "minimum": 1,
+              "enum": [
+                1,
+                2,
+                4,
+                6,
+                8,
+                16,
+                32,
+                64,
+                128,
+                256,
+                512,
+                1024,
+                2048,
+                4096,
+                8192,
+                16384,
+                32768,
+                65536
+              ],
+              "description": "Minimum size of the blocks in bytes to split and fragment a page block to assign to an allocation request.\\n It is used only by the POOLED memory allocator. The value has to be power of two. Default value is 16."
+            },
+            "page-size": {
+              "type": "integer",
+              "default": 4194304,
+              "minimum": 1,
+              "description": "Size of the page in bytes to allocate memory as a block. \\n It is used only by the POOLED memory allocator. Its default value is 1 << 22 (about 4 MB)"
+            },
+            "metadata-space-percentage": {
+              "type": "number",
+              "default": 12.5,
+              "minimum": 1,
+              "description": "Percentage of the allocated native memory that is used for the metadata such as indexes, offsets, etc. \\n It is used only by the POOLED memory allocator. Its default value is 12.5."
+            },
+            "persistent-memory-directory": {
+              "type": "string",
+              "description": "Specifies the directory where the non-volatile memory (e.g. Intel Optane) is mounted. \\n If this element is not defined, the RAM is used as a native memory. \\n This directory will be created automatically if it does not exist.\\n To avoid collisions, every member of the cluster will create its own subfolder to work with the non-volatile memory."
+            }
+          }
+        },
+        {
+          "anyOf": [
+            {
+              "not": {
+                "properties": {
+                  "allocator-type": {
+                    "const": "POOLED"
+                  }
+                }
+              }
+            },
+            {
+              "required": [
```

Comment:
I'm not sure these are necessary to make required. The comemnt in the yaml says that these are "used only by the POOLED allocator". But it doesn't mean they have to be explicitly specified in the config, since all of them have a default value. So if they are not specified explicitly then the default will be used.

Reply:
updated in general